File: 3451-string-compression-iii/string-compression-iii.java
class Solution {
    public String compressedString(String word) {
        StringBuilder sb = new StringBuilder();
        char prevChar = word.charAt(0);
        int prevCharCount = 1;

        for (int i = 1; i < word.length(); i++) {
            if (prevChar == word.charAt(i)) {
                prevCharCount++;
                if (prevCharCount == 9) {
                    sb.append(String.valueOf(prevCharCount));
                    sb.append(String.valueOf(prevChar));
                    prevCharCount = 0;
                }
            } else {
                if (prevCharCount > 0) {
                    sb.append(String.valueOf(prevCharCount));
                    sb.append(String.valueOf(prevChar));
                }
                prevCharCount = 1;
                prevChar = word.charAt(i);
            }
        }

        if (prevCharCount > 0) {
            sb.append(String.valueOf(prevCharCount));
            sb.append(String.valueOf(prevChar));
        }

        return sb.toString();

    }
}

File: 3451-string-compression-iii/string-compression-iii.cpp
class Solution {
public:
    string compressedString(string word) {
        string comp="";
        for(int i=0;i<word.length();i++){
            int cnt=1;
            while(word[i]==word[i+1] ){
                cnt++;
                if(cnt==9){
                    comp+=to_string(cnt);
            comp+=word[i];
            cnt=0;
                }
                i++;

            }
            if(cnt){
            comp+=to_string(cnt);
            comp+=word[i];}
        }
        return comp;
    }
};